packages/math-input/src/components/input/math-wrapper.ts

```diff
@@ -5,129 +5,84 @@
+    FRAC: (mathQuill) => {
+        mathQuill.cmd("\\frac");
     },
-    ["EQUAL"]: {str: "=", fn: ActionType.WRITE},
-    ["NEQ"]: {str: "\\neq", fn: ActionType.WRITE},
-    ["CDOT"]: {str: "\\cdot", fn: ActionType.WRITE},
-    ["PERCENT"]: {str: "%", fn: ActionType.WRITE},
-    ["LEFT_PAREN"]: {str: "(", fn: ActionType.CMD},
-    ["RIGHT_PAREN"]: {str: ")", fn: ActionType.CMD},
-    ["SQRT"]: {str: "sqrt", fn: ActionType.CMD},
-    ["PI"]: {str: "pi", fn: ActionType.CMD},
-    ["THETA"]: {str: "theta", fn: ActionType.CMD},
-    ["RADICAL"]: {str: "nthroot", fn: ActionType.CMD},
-    ["LT"]: {str: "<", fn: ActionType.WRITE},
-    ["LEQ"]: {str: "\\leq", fn: ActionType.WRITE},
-    ["GT"]: {str: ">", fn: ActionType.WRITE},
-    ["GEQ"]: {str: "\\geq", fn: ActionType.WRITE},
-    ["UP"]: {str: "Up", fn: ActionType.KEYSTROKE},
-    ["DOWN"]: {str: "Down", fn: ActionType.KEYSTROKE},
-    // The `FRAC_EXCLUSIVE` variant is handled manually, since we may need to do
-    // some additional navigation depending on the cursor position.
-    ["FRAC_INCLUSIVE"]: {str: "/", fn: ActionType.CMD},
-};
-
-const NormalCommands = {
-    ["LOG"]: "log",
-    ["LN"]: "ln",
-    ["SIN"]: "sin",
-    ["COS"]: "cos",
-    ["TAN"]: "tan",
-};
-
-const ArithmeticOperators = ["+", "-", "\\cdot", "\\times", "\\div"];
-const EqualityOperators = ["=", "\\neq", "<", "\\leq", ">", "\\geq"];
-
-const Numerals = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"];
-const GreekLetters = ["\\theta", "\\pi"];
-const Letters = [
-    "A",
-    "B",
-    "C",
-    "D",
-    "E",
-    "F",
-    "G",
-    "H",
-    "I",
-    "J",
-    "K",
-    "L",
-    "M",
-    "N",
-    "O",
-    "P",
-    "Q",
-    "R",
-    "S",
-    "T",
-    "U",
-    "V",
-    "W",
-    "X",
-    "Y",
-    "Z",
-];
-
-// We only consider numerals, variables, and Greek Letters to be proper
-// leaf nodes.
-const ValidLeaves = [
-    ...Numerals,
-    ...GreekLetters,
-    ...Letters.map((letter) => letter.toLowerCase()),
-    ...Letters.map((letter) => letter.toUpperCase()),
-];
-
-const KeysForJumpContext = {
-    [CursorContext.IN_PARENS]: "JUMP_OUT_PARENTHESES",
-    [CursorContext.IN_SUPER_SCRIPT]: "JUMP_OUT_EXPONENT",
-    [CursorContext.IN_SUB_SCRIPT]: "JUMP_OUT_BASE",
-    [CursorContext.BEFORE_FRACTION]: "JUMP_INTO_NUMERATOR",
-    [CursorContext.IN_NUMERATOR]: "JUMP_OUT_NUMERATOR",
-    [CursorContext.IN_DENOMINATOR]: "JUMP_OUT_DENOMINATOR",
+    JUMP_OUT_PARENTHESES: handleJumpOut,
+    JUMP_OUT_EXPONENT: handleJumpOut,
+    JUMP_OUT_BASE: handleJumpOut,
+    JUMP_INTO_NUMERATOR: handleJumpOut,
+    JUMP_OUT_NUMERATOR: handleJumpOut,
+    JUMP_OUT_DENOMINATOR: handleJumpOut,
+    LEFT: handleArrow,
+    RIGHT: handleArrow,
+    LOG: buildNormalFunctionCallback("log"),
+    LN: buildNormalFunctionCallback("ln"),
+    SIN: buildNormalFunctionCallback("sin"),
+    COS: buildNormalFunctionCallback("cos"),
+    TAN: buildNormalFunctionCallback("tan"),
 };
 
+// Notes about MathQuill
```

Comment:
I like this commentary, but IMHO, it'd be better to move this commentary on MathQuill up to the top of this file and move the comment at the top about the MathWrapper to here as class documentation.